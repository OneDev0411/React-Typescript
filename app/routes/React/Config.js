// Config.js
import React from 'react'
import { Route, IndexRoute, Redirect } from 'react-router'

// Main component
import App from '../../components/App'

// Pages
import Landing from '../../components/Pages/Landing'
import Recents from '../../components/Pages/Dashboard/Recents'
import Mls from '../../components/Pages/Dashboard/Mls'
import Contacts from '../../components/Pages/Dashboard/Contacts'
import Tasks from '../../components/Pages/Dashboard/Tasks'
import Transactions from '../../components/Pages/Dashboard/Transactions'
import SingleTransaction from '../../components/Pages/Dashboard/Transactions/Single'
import NewTransaction from '../../components/Pages/Dashboard/Transactions/New'
import SignUp from '../../components/Pages/SignUp'
import SignIn from '../../components/Pages/SignIn'
import Verify from '../../components/Pages/Verify'
import Password from '../../components/Pages/Password'
import NoMatch from '../../components/Pages/NoMatch'
import Settings from '../../components/Pages/Account/Settings'
import Notifications from '../../components/Pages/Account/Notifications'
import Recommend from '../../components/Pages/Dashboard/Mls/Listing/Recommend'
import Agents from '../../components/Pages/Dashboard/Mls/Agents'

export default (
  <Route path="/" component={App}>
    <IndexRoute component={Landing}/>
    <Route path="signup" component={SignUp}/>
    <Route path="signin" component={SignIn}/>
    <Route path="account/settings" component={Settings}/>
    <Route path="account/notifications" component={Notifications}/>
    <Redirect from="dashboard" to="dashboard/recents" />
    <Route path="dashboard/recents" component={Recents}/>
    <Route path="dashboard/recents/:room_id" component={Recents}/>
    <Route path="dashboard/mls" component={Mls}/>
    <Route path="dashboard/mls/agents" component={Agents}/>
    <Route path="dashboard/mls/listing/recommend" component={Recommend}/>
    <Route path="dashboard/contacts" component={Contacts}/>
    <Route path="dashboard/contacts/:id" component={Contacts}/>
    <Route path="dashboard/tasks" component={Tasks}/>
    <Route path="dashboard/transactions" component={Transactions}/>
    <Route path="dashboard/transactions/new" component={NewTransaction}/>
    <Route path="dashboard/transactions/:id" component={SingleTransaction}/>
    <Route path="dashboard/transactions/:id/attachments/:id" component={SingleTransaction}/>
    <Route path="verify/:slug" component={Verify}/>
    <Route path="password/:slug" component={Password}/>
    <Route path="*" component={NoMatch}/>
  </Route>
)