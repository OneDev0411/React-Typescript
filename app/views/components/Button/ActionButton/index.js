import styled from 'styled-components'
import PropTypes from 'prop-types'

const propTypes = {
  onClick: PropTypes.func
}

const defaultTypes = {
  onClick: () => {}
}

const ActionButton = styled.button`
  padding: 0.5em 1em;
  line-height: 1;
  border-width: 0;
  border-radius: 3px;
  color: ${props => (props.disabled ? '#333' : '#fff')};
  background: ${props => (props.disabled ? '#cecece' : '#2196f3')};

  ${props =>
    !props.disabled &&
    `&:hover {
    background: #107CEB;
  }`};
`

ActionButton.propTypes = propTypes
ActionButton.defaultProps = defaultTypes

export default ActionButton
