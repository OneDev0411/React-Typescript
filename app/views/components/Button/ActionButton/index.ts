import { HTMLProps } from 'react'

import { ButtonProps } from '@material-ui/core'
import styled, { css } from 'styled-components'

import { ButtonAppearances } from '../styles/ButtonAppearances'

export interface ActionButtonProps
  extends Omit<HTMLProps<HTMLButtonElement>, 'size' | 'ref' | 'type'> {
  /**
   * The appearance of the button.  {primary, outline, icon, link}
   */
  appearance?: keyof typeof ButtonAppearances

  /**
   * When true, the button is disabled.
   */
  disabled?: boolean

  /**
   * When true, the button is activated.
   */
  isActive?: boolean

  /**
   * When true, the button width is 100%.
   */
  isBlock?: boolean

  /**
   * DEPRECATED
   * When true, the button appearances should be ghost.
   */
  inverse?: boolean

  /**
   * The size of the button.
   */
  size?: 'small' | 'medium' | 'large' | 'XLarge'

  type?: ButtonProps['type']

  /**
   * The brand color of active team.
   */
  brandColor?: string

  noBorder?: boolean
  isOpen?: boolean

  rightRounded?: boolean // !
  leftRounded?: boolean // !
}

export const defaultProps: Partial<ActionButtonProps> = {
  type: 'button',
  appearance: 'primary',
  disabled: false,
  isActive: false,
  isBlock: false,
  inverse: false,
  size: 'medium',
  brandColor: ''
}

export const getSizeDependentStyles = props => {
  switch (props.size) {
    case 'small': {
      const size = 2.33

      return {
        height: `${size}em`,
        fontSize: '0.875rem',
        lineHeight: size
      }
    }

    case 'large': {
      const size = 2.39

      return {
        height: `${size}em`,
        fontSize: '1.125rem',
        lineHeight: size
      }
    }

    default: {
      const size = 2.375 // MUI button height / 1rem

      return {
        height: `${size}em`,
        fontSize: '1rem',
        lineHeight: size
      }
    }
  }
}

export const getAppearance = props => {
  let appearance = props.appearance

  return css`
    ${ButtonAppearances[appearance]};
    ${getSizeDependentStyles(props)};
    padding: 0 1rem;
  `
}

const Button = styled.button<ActionButtonProps>(getAppearance)

Button.defaultProps = defaultProps

export default Button
