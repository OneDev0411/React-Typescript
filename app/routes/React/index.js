// index.js
import React from 'react'
import { match, RoutingContext } from 'react-router'
import ReactDOMServer from 'react-dom/server'

// Store
import AppStore from '../../stores/AppStore'

// Config
import routes from './Config'

module.exports = (app, config) => {

  app.get('*',(req, res) => {

    match({ routes, location: req.url }, (error, redirectLocation, renderProps) => {
      
      let reactMarkup = ReactDOMServer.renderToStaticMarkup(<RoutingContext data={AppStore.data} {...renderProps} />)
      
      res.locals.reactMarkup = reactMarkup

      console.log(reactMarkup)

      if (error) {
      
        res.status(500).send(error.message)
      
      } else if (redirectLocation) {
        
        res.redirect(302, redirectLocation.pathname + redirectLocation.search)

      } else if (renderProps) {
        
        // Success!
        res.status(200).render('index.html')
      
      } else {

        res.status(404).render('index.html')
      
      }

    })
  })
  
}