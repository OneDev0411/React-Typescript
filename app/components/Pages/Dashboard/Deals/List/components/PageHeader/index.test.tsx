import React from 'react'
import { shallow } from 'enzyme'
import toJson from 'enzyme-to-json'

import ActionButton from 'components/Button/ActionButton'

import { Header } from '.'

describe('Deal list page header', () => {
  it('renders the page header', () => {
    const wrapper = shallow(
      <Header
        user={{}}
        title=""
        isSideMenuOpen
        onMenuTriggerChange={() => null}
      />
    )

    expect(toJson(wrapper)).toMatchSnapshot()
  })

  it('renders create deal button when side bar is open', () => {
    const wrapper = shallow(
      <Header
        user={{}}
        title=""
        isSideMenuOpen
        onMenuTriggerChange={() => null}
      />
    )

    expect(wrapper.find(ActionButton)).toHaveLength(1)
  })

  it('renders create deal button when side bar is closed', () => {
    const wrapper = shallow(
      <Header
        user={{}}
        title=""
        isSideMenuOpen
        onMenuTriggerChange={() => null}
      />
    )

    expect(wrapper.find(ActionButton)).toHaveLength(1)
  })
})
