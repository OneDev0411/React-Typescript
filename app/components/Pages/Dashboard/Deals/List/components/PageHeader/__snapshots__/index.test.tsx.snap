// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Deal list page header renders the page header 1`] = `
<withRouter(PageHeader)
  isFlat={true}
>
  <PageTitle
    showBackButton={false}
  >
    <Trigger
      className="c-menu-trigger"
      isExpended={true}
      onClick={[Function]}
      style={Object {}}
      tooltip={null}
    />
    <headings__H1 />
  </PageTitle>
  <PageMenu>
    <ExportDeals
      user={Object {}}
    />
    <WithStyles(ForwardRef(Button))
      color="secondary"
      data-test="create-deal-button"
      onClick={[Function]}
      style={
        Object {
          "marginLeft": "1rem",
        }
      }
      variant="contained"
    >
      Create New Deal
    </WithStyles(ForwardRef(Button))>
  </PageMenu>
</withRouter(PageHeader)>
`;
